cosmos/scheduler/cmd/scheduler.go

```diff
@@ -31,47 +33,61 @@ type Scheduler struct {
 // Run starts the Cosmos Scheduler
 func (s *Scheduler) Run() {
 	log.Println("starting cosmos scheduler")
-	quit := make(chan bool)
 	overlayUpdates, err := s.watchOverlayUpdates()
 	if err != nil {
 		log.Println(err)
 		return
 	}
+	// safely create or update secret on startup
+	go s.configureIngestionSecret(true)
+	go s.checkMongoConnection()
+	go s.watchBucketUpdates()
 	for {
-		go s.watchBucketUpdates(quit)
+		// blocks until an overlay update is recieved on the overlayUpdates channel
 		<-overlayUpdates
 		log.Println("received an overlay update")
-		quit <- true
 	}
 }
 
-func (s *Scheduler) configureIngestionSecret() error {
+func (s *Scheduler) checkMongoConnection() {
+	ticker := time.NewTicker(5 * time.Second)
+	ctx := context.Background()
+	for {
+		<- ticker.C
+		err := s.MongodbClient.Ping(ctx, readpref.Nearest())
+		if err != nil{
+			log.Panicln("ping healthcheck failed:", err)
+		}
+	}
+}
+
+func (s *Scheduler) configureIngestionSecret(init bool) {
 	accessKey, secretKey, err := s.Pensieve.GetServiceAccountCredentials("md-ingestion")
 	if err != nil {
-		return err
+		log.Panicln("error getting service account credentials")
```

Comment:
Panics are better suited for programming errors. This is more Fatal material.

Reply:
Yeah I was just reading about that. I think I'll need to change a few of these.